### PR TITLE
feat(server): abandon cell outputs after the block has been abandoned

### PR DESCRIPTION
**Pull request by shaojunda**

### CODE REVIEW

File: server/test/controllers/api/v1/cell_output_lock_scripts_controller_test.rb [hunk 1]
```diff
@@ -4,31 +4,35 @@ module Api
   module V1
     class CellOutputLockScriptsControllerTest < ActionDispatch::IntegrationTest
       test "should get success code when call show" do
-        cell_output = create(:cell_output, :with_full_transaction)
+        block = create(:block, :with_block_hash)
```
Comment (classicalliu) [inline server/test/controllers/api/v1/cell_output_lock_scripts_controller_test.rb]: Why not initialize in `setup` ?

File: server/app/models/block.rb [hunk 2]
```diff
@@ -57,6 +58,7 @@ def authenticate!
   def abandon!
     update!(status: "abandoned")
     ChangeCkbTransactionsStatusWorker.perform_async(id, "abandoned")
+    ChangeCellOutputsStatusWorker.perform_async(id, "abandoned")
```
Comment (classicalliu) [inline server/app/models/block.rb]: What's mean of `abandoned` ?

Comment (shaojunda) [reply server/app/models/block.rb]: 弃用或者废弃，既不是 live 也不是 dead 的一种状态。

File: server/test/controllers/api/v1/cell_output_lock_scripts_controller_test.rb (hunk 1 above)
Comment (shaojunda) [reply server/test/controllers/api/v1/cell_output_lock_scripts_controller_test.rb]: `setup` will runs before every test. but there is one test no need to create cell output.

Comment (classicalliu) [reply server/test/controllers/api/v1/cell_output_lock_scripts_controller_test.rb]: I see a lot of those two lines in this file, and if you don't need in every test case, also can use lazy load.

File: server/app/models/block.rb (hunk 2 above)
Comment (classicalliu) [reply server/app/models/block.rb]: ok

File: server/test/controllers/api/v1/cell_output_lock_scripts_controller_test.rb (hunk 1 above)
Comment (shaojunda) [reply server/test/controllers/api/v1/cell_output_lock_scripts_controller_test.rb]: Thank you, I will fix this.

Comment (shaojunda) [reply server/test/controllers/api/v1/cell_output_lock_scripts_controller_test.rb]: @classicalliu I don't like use lazy load in test, so I solved this problem with the `create_cell_output ` method. please review again.

**Pull request merged**